internal/ingress/controller/template/template.go

```diff
@@ -1672,3 +1673,42 @@ func convertGoSliceIntoLuaTable(goSliceInterface interface{}, emptyStringAsNil b
 		return "", fmt.Errorf("could not process type: %s", kind)
 	}
 }
+
+// buildOriginRegex returns an origin as a regex
+func buildOriginRegex(origin string) string {
+	if !strings.HasPrefix(origin, "*") {
+		return origin
+	}
+
+	origin = strings.Replace(origin, "*.", "", 1)
+	parts := strings.Split(origin, ".")
+
+	return "(" + strings.Join(parts, "\\.") + ")"
+}
+
+// buildCorsOriginRegex builds the regex string required by nginx
+func buildCorsOriginRegex(origins string) string {
+	var originSplitList = strings.Split(origins, ",")
+	if searchCorsOrigins(originSplitList, "*") {
+		return "set $http_origin *;\nset $cors 'true';"
+	}
+
+	var originsRegex string = "if ($http_origin ~* .*("
```

Comment:
Why are we matching `.*` first? Doesn't that mean that if I trust `facebook.com`, then `maliciousfacebook.com` will also be trusted?

Reply:
I just tried this one a regextester.com (there could be other reliable sources but no. It doesn't match.
If you'd like to test: `.*((http:\/\/facebook.com)|(http:\/\/google.com))` this is the regex string
![image](https://user-images.githubusercontent.com/3633214/132773332-ce1fd15c-2310-41f1-a723-e1a7411bb568.png)

If it matched, the origin text would be in blue.
![image](https://user-images.githubusercontent.com/3633214/132773430-b421bf3f-b1f9-4290-b165-d59d712107d0.png)

Reply:
i removed it, you were right